### DISCUSSION TITLE
03/02/2022 bug

### DISCUSSION THREAD

**Discussion by posssson** [doudar/SmartSpin2k#323] [Q&A]
Hello,

The new version doesn't work for me. I can connect my assioma but it is always 0 watts. and debug page is freeze. 

And an easy solution for wifi connection problem is to set an other param and dissociate bluetooth and wifi name. It works perfectly for me.

**Comment by doudar** [comment]
You’ll probably have to update the data files on your SS2K as well.

If you’d like to try 2.2.8, it’s been released. I use Assiomas and haven’t had any issues, is there a chance it’s paired to your right pedal?

**Reply by posssson** [reply]
Impossible to connect to internet. Sometimes my pedals work sometimes no don't know why :(. I must quit Zwift. I think I will develop an app to control without zwift. But nice work :)

**Reply by doudar** [reply]
Feel free to use and expand on the custom characteristic for this:

https://github.com/doudar/SmartSpin2k/blob/develop/CustomCharacteristic.md

https://github.com/doudar/SmartSpin2k/blob/d0d6a524ddd10927f569f89a7e77cc8793693ce7/src/BLE_Server.cpp#L621-L957

**Reply by eMadman** [reply]
@posssson is it possible for you to manually flash 2.2.8 and then reset the firmware?  You can do this over USB without a wifi connection:

https://github.com/doudar/SmartSpin2k/wiki/Loading-Software

The update has some bluetooth improvements and might help with your connectivity issue.  

If the issue continues after you apply the update, could you provide logs from your SS2K?    You can get this from "SmartSpin Debugging Info" screen or with the UDP logger.

**Reply by posssson** [reply]
I flash evertimes my SS2K without internet with vscode. I reset before to erase all data on it. Yes if the problem continues I will send u the logs.

**Reply by MarkusSchneider** [reply]
@posssson New version worked for me and is really stable. I'm using Assioma DUO.
What exactly do you mean with "Impossible to connect to internet". 
Is it not possible to connect to http://smartspin2k.local via Wifi or does SS2k not send Power Values to Zwift?

Could it be a problem with your ESP32 controller?

I had also connection problems in the past. My problem was a weak BLE connection between SS2k and my notebook.
I adjusted the position of bike and notebook and all is now running very stable. I checked it with a HR Belt connected directly to zwift via BLE.
You can try to use your mobile phone with the zwift companion app and connect ss2k via your mobile.
https://support.zwift.com/en_us/using-the-zwift-companion-app-Hybn8qzPr
